app/code/Magento/CatalogImportExport/Model/Import/Product/LinkProcessor.php

```diff
@@ -0,0 +1,340 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\CatalogImportExport\Model\Import\Product;
+
+use Magento\Catalog\Model\ResourceModel\Product\LinkFactory;
+use Magento\CatalogImportExport\Model\Import\Product;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\ImportExport\Model\Import;
+use Magento\ImportExport\Model\ResourceModel\Helper;
+use Magento\ImportExport\Model\ResourceModel\Import\Data;
+use Psr\Log\LoggerInterface;
+
+class LinkProcessor
+{
+    /**
+     * @var \Magento\CatalogImportExport\Model\Import\Product
+     */
+    protected $_entityModel;
+
+    /**
+     * Links attribute name-to-link type ID.
+     *
+     * @var array
+     */
+    protected $_linkNameToId = [];
+
+    /**
+     * @var LinkFactory
+     */
+    protected $_linkFactory;
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $_logger;
+
+    /**
+     * @var Helper
+     */
+    protected $_resourceHelper;
+
+    /**
+     * @var Data
+     */
+    protected $_dataSourceModel;
+
+    /**
+     * @var SkuProcessor
+     */
+    protected $skuProcessor;
+
+    /**
+     * @var AdapterInterface
+     */
+    protected $_connection;
```

Comment:
I think it'd be great to extract the "resource model" to encapsulate database operations and deload this class a bit

Reply:
I don't understand - can you elaborate?

Reply:
I mean it would be good to extract all database operations to a separate class - it will reduce class size and coupling between objects.

Reply:
wondering how I will inject $this->resource into the resourcemodel ...